docs/src/main/tut/auth.md

```diff
@@ -0,0 +1,139 @@
+---
+layout: default
+title: Authentication
+---
+## Authentication
+
+A [service] is a `Kleisli[Task, Request, Response]`, the composable version of
+`Request => Task[Response]`. http4s provides an alias called `Service[Request,
+Response]`. A service with authentication also requires some kind of `User`
+object which identifies which user did the request. To store the `User` object
+along with the `Request`, there's `AuthedRequest[User]`, which is equivalent to
+`(User, Request)`. So the service has the signature `AuthedRequest[User] =>
+Task[Response]`, or `AuthedService[User]`. So we'll need a `Request => User`
+function, or more likely, a `Request => Task[User]`, because the `User` will
+come from a database. Which is a `Service[Request, User]`, which we can compose
+with the `AuthedService[User]` to get a [service]. Or in code:
+
+```tut:book
+import scalaz._, Scalaz._, scalaz.concurrent.Task
+import org.http4s._
+import org.http4s.dsl._
+
+case class User(id: Long, name: String)
+
+// Needs to be Kleisli for the type inference of ||| to work.
+val authUser: Kleisli[Task, Request, User] = Kleisli(_ => Task.delay(???))
+val authedService: AuthedService[User] =
+  AuthedService {
+    case GET -> Root / "welcome" as user => Ok(s"Welcome, ${user.name}")
+  }
+val service: HttpService = authedService <=< AuthedRequest(authUser)
+```
+
+## Returning an error Response
+
+Usually, it should also be possible to send back a 403 in case there was no
+valid login. The 403 response can be adjusted as needed, some applications use a
+redirect to a login page, or a popup requesting login data. With the upcoming of
+[SPA], the correct http status codes are relevant again.
+
+### With Kleisli
+
+To allow for failure, the `authUser` function has to be adjusted to a `Request
+=> Task[String \/ User]`. So we'll need to handle that possibility. For advanced
+error handling, we recommend an error [ADT] instead of a `String`.
+
+```tut:book
+val authUser: Kleisli[Task, Request, String \/ User] = Kleisli(_ => Task.delay(???))
+
+// Needs to be Kleisli for the type inference of ||| to work.
+val onFailure: Kleisli[Task, AuthedRequest[String], Response] = Kleisli(req => Forbidden(req.authInfo))
```

Comment:
Maybe out of scope for this tut, but I think this is more likely to be a 401 than a 403.  It could be either, and I guess that gets into the error ADT.